valhalla/sif/dynamiccost.h

```diff
@@ -605,7 +605,7 @@ class DynamicCost {
         if (access_type == baldr::AccessType::kTimedAllowed)
           time_allowed = true;
 
-        if (current_time == 0) {
+        if (current_time == 0 || ignore_common_restrictions_) {
```

Comment:
actually (and this would be fixing a bug): we should add smth similar to the `Restricted()` function further up which looks at complex TRs. it could use the new internal `ignore_turn_restrictions_` and early abort